File: Others/medium/SunsetViews.py
def sunsetViews(buildings, direction):
    if not buildings:
        return []

    if direction == "WEST":
        dr = 1
        curr_start = 0
        curr_end = len(buildings)
    else:
        dr = -1
        curr_start = len(buildings) - 1
        curr_end = -1

    sol = [curr_start]
    max_h = buildings[curr_start]

    while curr_start != curr_end:
        if buildings[curr_start] > max_h:
            if dr == 1:
                sol.append(curr_start)
            else:
                sol.insert(0, curr_start)
        max_h = max(max_h, buildings[curr_start])
        curr_start += dr

    return sol

File: Golang/medium/sunset-views.go
package main

func SunsetViews(buildings []int, direction string) []int {
	if len(buildings) == 0 {
		return []int{}
	}

	if direction == "EAST" {
		res := []int{len(buildings) - 1}
		for idx := len(buildings) - 2; idx >= 0; idx-- {
			last := res[len(res)-1]
			if buildings[idx] > buildings[last] {
				res = append(res, idx)
			}
		}

		for idx := 0; idx < len(res)/2; idx++ {
			res[idx], res[len(res)-idx-1] = res[len(res)-idx-1], res[idx]
		}

		return res

	} else {
		res := []int{0}
		for idx := 1; idx < len(buildings); idx++ {
			last := res[len(res)-1]
			if buildings[idx] > buildings[last] {
				res = append(res, idx)
			}
		}
		return res
	}
}
